src/main/java/com/arpnetworking/metrics/incubator/impl/TsdPeriodicMetricsFactory.java

```diff
@@ -0,0 +1,251 @@
+import com.arpnetworking.metrics.impl.TsdMetricsFactory;
+import com.arpnetworking.metrics.incubator.PeriodicMetricsFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public final class TsdPeriodicMetricsFactory implements PeriodicMetricsFactory {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void registerPolledMetric(final Consumer<Metrics> consumer) {
+        _polledMetricsRegistrations.add(consumer);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void record(final Consumer<Metrics> consumer) {
+        _currentPeriodicMetrics.readLocked(metrics -> consumer.accept(new NonClosingMetrics(metrics)));
+    }
+
+    private void cyclePeriodMetrics() {
+        _currentPeriodicMetrics.readLocked(this::recordPolledMetrics);
+        final Metrics metrics = _currentPeriodicMetrics.getAndSetReference(_metricsFactory.create());
+        metrics.close();
+    }
+
+    private void recordPolledMetrics(final Metrics metrics) {
+        for (Consumer<Metrics> polledMetric : _polledMetricsRegistrations) {
+            polledMetric.accept(metrics);
+        }
+    }
+
+    private TsdPeriodicMetricsFactory(final Builder builder) {
+        _metricsFactory = builder._metricsFactory;
+
+        _currentPeriodicMetrics = new SafeRefLock<>(_metricsFactory.create());
+
+        _closingExecutor.scheduleAtFixedRate(this::cyclePeriodMetrics, 500, 500, TimeUnit.MILLISECONDS);
+    }
+
+    private final MetricsFactory _metricsFactory;
+    private final SafeRefLock<Metrics> _currentPeriodicMetrics;
+    private final ScheduledExecutorService _closingExecutor = new ScheduledThreadPoolExecutor(1, (r) -> new Thread(r, "MetricsCloser"));
+    private final Set<Consumer<Metrics>> _polledMetricsRegistrations = ConcurrentHashMap.newKeySet();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TsdPeriodicMetricsFactory.class);
+
+    /**
+     * Implementation of the Builder pattern for the {@link TsdPeriodicMetricsFactory} class.
+     *
+     * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+     */
+    public static final class Builder implements com.arpnetworking.commons.builder.Builder<TsdPeriodicMetricsFactory> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            this(LOGGER);
+        }
+
+        // NOTE: Package private for testing
+        /* package private */ Builder(@Nullable final Logger logger) {
```

Comment:
Following pattern established elsewhere.  But I think that was erroneous.  Fixed.